spec/support/password_client.rb

```diff
@@ -10,6 +10,8 @@ def initialize(password: 'GOOD')
   end
 
   def create_authorization(params = {})
+    raise Octokit::UnprocessableEntity if params[:note].nil?
```

Comment:
I'm sorry I'm lost why are you raising this exception here and then just handling it with nothing later?

Reply:
I'm raising it here because that's what GitHub does.

However, this is one of those times:

1.  Encounter bug
2.  Fix bug 1
3.  Encounter bug
4.  Fix bug 3 in a way that makes bug 1 irrelevant.

That said, this mock is a little more accurate _with_ this line than without it (could be better).

In cases where the user _has_ 2FA setup, the new "dummy" attempt at creating an authorization raises an `Octokit::OneTimePasswordRequired`. But in cases where they _don't_ it raises an `Octokit::UnproccesableEntity` because the `note` is required.

Reply:
really it should be something like:

```ruby
if @password == 'TFA' && !params.dig(:headers, 'X-Github-OTP')
  raise Octokit::OneTimePasswordRequired
end

if @password == 'GOOD' || @password == 'TFA'
  if params[:note].nil?
    raise Octokit::UnprocessableEntity
  else
    OpenStruct.new(token: '*')
  end
else
    raise Octokit::Unauthorized
end
```

But...

Reply:
Nope, still not right.